dashboard/app/models/user.rb

```diff
@@ -355,13 +357,21 @@ def save_parent_email_preference
 
   # Enables/disables sharing of emails of teachers in the U.S. to Code.org regional partners based on user's choice.
   def save_email_reg_partner_preference
+    puts "inside save_email_reg_partner_preference"
     user = User.find_by_email_or_hashed_email(email)
     if teacher? && share_teacher_email_reg_partner_opt_in_radio_choice.casecmp?("yes")
       user.share_teacher_email_regional_partner_opt_in = DateTime.now
       user.save!
     end
   end
 
+  # Puts teachers directly into the progress table v2 view when new account is created.
+  def save_show_progress_table_v2
+    if teacher?
+      self.show_progress_table_v2 = true
+    end
```

Comment:
@etaderhold I had to remove the `save!` here to get it to pass drone.  With `save!` multiple tests failed - specifically tests related to calling `update_school_info` - which is odd because i had not touched anything there.  That being said, without `saving`, everything appears to work.  I think this is because we are saving after this method is called - but I cannot find out where.

Reply:
Note to my future self: We went with removing the save but then changing this to run `before_create` to ensure the changes did get saved at some point. This shows the order in which[ these callbacks occur](https://api.rubyonrails.org/v7.1.3.4/classes/ActiveRecord/Callbacks.html).